firewall.tf

```diff
@@ -14,3 +14,12 @@ resource "azurerm_postgresql_virtual_network_rule" "virtualnetworks" {
   server_name         = azurerm_postgresql_server.server.name
   subnet_id           = each.value
 }
+
+resource "azurerm_postgresql_firewall_rule" "public" {
```

Comment:
This should be explicitly configurable. A default of "if no ips are allowed, all ips are allowed" isn't good IMHO as a user could just as well only allow (internal) subnets to access the server and no external ip at all.

Also, this rule can be achieved by setting the allowed ips to one item ranging from 0.0.0.0 to 255.255.255.255.